### ISSUE TITLE
issue with pairwise comparisons

### ISSUE THREAD

**Issue by amm1623** [PombertLab/SYNY#7]
Hi!

Thank you for this program that is pretty straightforward.

I am trying to run it on a reference genome from genbank that does have a gbff file and my assembled genome for which I have a gff3 file and fasta. I generated a gbff from these using a custom script but I am not sure how well this worked.
When I run syny, the output of the comparison between my genome to the reference has plots but the other way around is blank. plots that require a pairwise comparison are empty. I was wondering if there is a way to input the gff3 and fasta file as I think this might be causing the issue.

Thank you!

**Comment by Pombert-JF** [comment]
The GFF3 annotation format is currently not supported as it lacks sequence data. I'm thinking about writing a GFF3 + FASTA to GBFF converter, but that will likely take a while before it is ready.

As for why your analyses fail, hard to tell without seeing your input/output files, but yes a non-GenBank compliant GBFF files would likely break down SYNY, since we extract the annotation and sequence data from it.

If your conversion script does not generate the `/translation=""` fields in the GBFF files, that would definitely break protein cluster analyses. Analyses based on genome alignments should work properly as long as the genome sequences are included in the GBFF files.

You could try converting GFF3 annotations to TBL format, then creating NCBI-compliant GBFF files 
from the TBL files with NCBI's table2asn. I believe that [funannotate](https://funannotate.readthedocs.io/en/latest/utilities.html) has a GFF3 to TBL conversion tool but I have not tested it.

If you don't mind sending me your GBFF file, I could have a look to pinpoint what the issue(s) are.

Best,
JF

**Comment by amm1623** [comment]
Thank you for the reply. That is definitely the problem. In comparing my gbff to the reference assembly it does not look correct and it is missing the translation fields. I will try the steps above. Thank you again.

**Comment by Pombert-JF** [comment]
I pushed a quick GFF3 + FASTA to GBFF converter to the Utils subdirectory. It is available here: [gff3_to_gbff.pl](https://github.com/PombertLab/SYNY/blob/main/Utils/gff3_to_gbff.pl). I tested it with NCBI GFF3 files.

The tool expects:
 - the same file prefixes: e.g. genome_1.fna  /  genome_1.gff
 - gene/mRNA/exon/CDS entries in the gff3 `type` column
 - ID and Parent tags in the `attributes` column

The GBFF files created do not adhere exactly to the GBFF format but work with SYNY. I did not have the chance to test it with files containing multiple isoforms however.

If the tool doesn't work with your data, I would need to see the structure of your GFF3 file(s) to figure out the issues.

Best,
JF
